go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -43,6 +43,13 @@ func newAddressManager(nodeName string, k kube.Interface, config *managementPort
 	return mgr
 }
 
+// gets IPs from node address manager
+func (c *addressManager) getAddr() *sets.String {
+	c.Lock()
+	defer c.Unlock()
+	return &c.addresses
```

Comment:
It needs to return a copy of the set

Reply:
you know what?! I am starting to act really dumb these days.. thanks for catching this, I don't know what I was thinking to simply play with locks just to return the same set LOL. I mean what's the point in me doing what I just did....

Reply:
no big deal I do stuff like this all the time :)

Reply:
oh casey already added it! func (c *addressManager) ListAddresses() []net.IP {
haha found this as I was reviewing your PR,! I just need to reuse that...